backend/config.ts

```
@@ -0,0 +1,53 @@
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+// Ensure authSource=admin is used for MongoDB authentication
+let MONGODB_URI = process.env.MONGODB_URI as string;
+if (MONGODB_URI && !MONGODB_URI.includes('authSource=')) {
+  MONGODB_URI += '?authSource=admin';
+} else if (MONGODB_URI && MONGODB_URI.includes('authSource=cost')) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)